scripts/write-env-vars-file.ts

```diff
@@ -6,6 +6,12 @@ dotenv.config();
 // Read the .env file
 const envVars = dotenv.parse(fs.readFileSync('.env'));
 
+Object.keys(envVars).map((envVarName) => {
+  if (envVarName.startsWith('ETHERSCAN_API_KEY')) {
+    delete envVars[envVarName];
+  }
+});
+
```

Comment:
I wouldn't mutate the envVars object and taking another look at this script I think it would be better to use a more functional approach: 

```
import * as dotenv from 'dotenv';
import * as fs from 'fs';

dotenv.config();

// Read the .env file
const envVars = dotenv.parse(fs.readFileSync('.env')) as { [key: string]: string };

// Filter and format the environment variables
const formattedEnvVars = Object.keys(envVars)
  .filter((key) => !key.startsWith('ETHERSCAN_API_KEY'))
  .reduce((acc, key) => {
    acc[key] = `\${env:${key}, ""}`;
    return acc;
  }, {} as { [key: string]: string });

// Convert formattedEnvVars to YAML format
const yamlEnvVars = Object.entries(formattedEnvVars)
  .map(([key, value]) => `${key}: ${value}`)
  .join('\n');

// Write YAML to env-vars.yml file
fs.writeFileSync('env-vars.yml', yamlEnvVars);
```

Reply:
Commited